### DISCUSSION TITLE
Is it possible to have multi word text match search?

### DISCUSSION THREAD

**Discussion by pycui** [milvus-io/milvus#38643] [Q&A and General discussion] [answered]
we are using https://milvus.io/docs/keyword-match.md#Text-Match feature, which we prefer over Full text match using BM25 since the exact control of filter. 
However, it seems currently it only supports per word filtering ,e.g. if I want to search "sample text 1", the best thing I can do is `filter = "TEXT_MATCH(text, 'sample') and TEXT_MATCH(text, 'text') and TEXT_MATCH(text, '1')"`, but this doesn't guarantee the order of words. 
Wondering if real "text match" is possible in a feature now?

**Comment by yhmo** [comment, marked as answer]
Currently the build-in tokenizers could not distinct phrases, maybe we can add more configurations to keep phrases for match in future versions.

A possible workaround is infix filtering:
`filter = 'text like "%sample text 1%"'`

**Comment by xiaofan-luan** [comment]
> filtering ,e.g. if I want to search "sample text 1", the best thing I can do is `filter = "TEXT_MATCH(text, 'sample') and TEXT_MATCH(text, 'text') and TEXT_MATCH(text, '1')"`, but this doesn't guarantee the order of words.
> Wondering if real "text match" is possible in a feature now?

Do you wanto match text or 1 or do you want to match text1 as one single phrase?

**Comment by xiaofan-luan** [comment]
if it's the previous one then
<img width="1046" alt="image" src="https://github.com/user-attachments/assets/a7e6610c-3f8f-476d-a6e8-83cdb76bd169" />

otherwise what you need is match phrase feature

We are working on it but it will be released on 2.6
